awx/ui_next/src/setupProxy.js

```diff
@@ -29,12 +32,4 @@ module.exports = app => {
         req.originalUrl.includes('login'),
     })
   );
-  app.use(
```

Comment:
I could only get websocket proxying to work if I configured it in package.json. I believe it has something to do with _when_ this configuration gets applied to the webpack dev server. Configuration from `package.json` gets added [when the server is created](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/start.js#L132), while `setupProxy.js` middleware gets added [later](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/webpackDevServer.config.js#L121).

For some reason this seems to matter. I'd love to get it working using `setupProxy.js` middleware but I haven't had much luck getting it to work after spending an hour so looking into it.

Reply:
does target in here update based on what you set the proxy to in package.json?  trying to figure out how you'd set it to proxy on a different url (if any changes would be needed in here)

Reply:
To use a new url, pass the new host and port values in as environment variables `TARGET_HOST`, `TARGET_PORT` _and_ update the package.json with the new `/websocket` url. 

In an ideal world, setting the environment vars would be _all_ that is needed but adding middleware to `setupProxy.js` doesn't seem to work correctly for websockets. The only workaround I could find is to pass the websocket url in as the `proxy` value in `package.json`, which causes the dev server to do the right thing.